plenum/server/replica.py

```diff
@@ -2429,7 +2430,8 @@ def caught_up_till_3pc(self, last_caught_up_3PC):
         self._remove_till_caught_up_3pc(last_caught_up_3PC)
         self._remove_ordered_from_queue(last_caught_up_3PC)
         self.checkpoints.clear()
-        self._remove_current_view_stashed_checkpoints(last_caught_up_3PC)
+        if last_caught_up_3PC[0] == self.viewNo:
```

Comment:
What if haven'e yet ordered anything in this view and last_ordered points to the previous view?

Reply:
This means that no stashed received checkpoints from this view must be removed. So we will not get into this `if` branch.